service/frontend/nexus_handler.go

```diff
@@ -319,7 +410,65 @@ func (h *nexusHandler) CancelOperation(ctx context.Context, service, operation,
 	return fmt.Errorf("unhandled response outcome: %T", response.GetOutcome()) //nolint:goerr113
 }
 
-// convertNexusHandlerError converts any 5xx user handler error to a downsream error.
+func (h *nexusHandler) forwardCancelOperation(
+	ctx context.Context,
+	service string,
+	operation string,
+	id string,
+	options nexus.CancelOperationOptions,
+	oc *operationContext,
+) error {
+	options.Header[dcRedirectionApiHeaderName] = "true"
+
+	client, err := h.nexusClientForActiveCluster(oc, service)
+	if err != nil {
+		return err
+	}
+
+	handle, err := client.NewHandle(operation, id)
+	if err != nil {
+		oc.logger.Warn("invalid Nexus cancel operation.", tag.Error(err))
+		return nexus.HandlerErrorf(nexus.HandlerErrorTypeBadRequest, "invalid operation")
+	}
+
+	err = handle.Cancel(ctx, options)
+	if err != nil {
+		oc.logger.Error("received error from remote cluster for forwarded Nexus cancel operation request.", tag.Error(err))
+		oc.metricsHandler = oc.metricsHandler.WithTags(metrics.NexusOutcomeTag("forwarded_request_error"))
+		return commonnexus.HandlerErrorFromClientError(err)
+	}
+
+	return nil
+}
+
+func (h *nexusHandler) nexusClientForActiveCluster(oc *operationContext, service string) (*nexus.Client, error) {
+	httpClient, err := h.forwardingClients.Get(oc.namespace.ActiveClusterName())
+	if err != nil {
+		oc.logger.Error("failed to forward Nexus request. error creating HTTP client", tag.Error(err), tag.SourceCluster(oc.namespace.ActiveClusterName()), tag.TargetCluster(oc.namespace.ActiveClusterName()))
+		oc.metricsHandler = oc.metricsHandler.WithTags(metrics.NexusOutcomeTag("request_forwarding_failed"))
+		return nil, nexus.HandlerErrorf(nexus.HandlerErrorTypeInternal, "request forwarding failed")
+	}
+
+	baseURL, err := url.JoinPath(
+		httpClient.Address,
```

Comment:
This is supposed to just be the address, not the base URL, you'll want to add the scheme (`http` / `https` based on whether TLS is enabled).

I noticed Michael mistakenly put the full URL there.

Reply:
Okay, I will address in a followup since I think that will require some refactoring in cluster metadata.